java/KinesisConnectors/src/main/java/com/amazonaws/services/msf/StreamingJob.java

```diff
@@ -44,39 +48,54 @@ private static Map<String, Properties> loadApplicationProperties(StreamExecution
         }
     }
 
-    private static KinesisStreamsSource<String> createKinesisSource(Properties inputProperties) {
+    private static <T> KinesisStreamsSource<T> createKinesisSource(Properties inputProperties, final DeserializationSchema<T> deserializationSchema) {
         final String inputStreamArn = inputProperties.getProperty("stream.arn");
-        return KinesisStreamsSource.<String>builder()
+        return KinesisStreamsSource.<T>builder()
                 .setStreamArn(inputStreamArn)
                 .setSourceConfig(Configuration.fromMap(Maps.fromProperties(inputProperties)))
-                .setDeserializationSchema(new SimpleStringSchema())
+                .setDeserializationSchema(deserializationSchema)
                 .build();
     }
 
-    private static KinesisStreamsSink<String> createKinesisSink(Properties outputProperties) {
+    private static <T> KinesisStreamsSink<T> createKinesisSink(Properties outputProperties, final SerializationSchema<T> serializationSchema) {
         final String outputStreamArn = outputProperties.getProperty("stream.arn");
-        return KinesisStreamsSink.<String>builder()
+        return KinesisStreamsSink.<T>builder()
                 .setStreamArn(outputStreamArn)
                 .setKinesisClientProperties(outputProperties)
-                .setSerializationSchema(new SimpleStringSchema())
+                .setSerializationSchema(serializationSchema)
                 .setPartitionKeyGenerator(element -> String.valueOf(element.hashCode()))
                 .build();
     }
 
+    private static <T> DataStream<T> createDataStream(final StreamExecutionEnvironment env, TypeInformation<T> typeInformation, KinesisStreamsSource<T> source) {
```

Comment:
It's just a matter of code style, but I think putting this code in the main directly is clearer.
The goals of these examples are 1/ code clarity, 2/ show best practices.
Inlining this code improve (1) and does not harm (2)